pyramid_oereb/standard/hook_methods.py

```diff
@@ -107,9 +106,9 @@ def get_symbol(request):
             class_='LegendEntry'
         ))
         legend_entry = session.query(model).filter(
-            model.type_code == type_code,
-            cast(model.legend_text, Text) == cast(cast(
-                json.loads(base64.b64decode(legend_text).decode('utf-8')), JSONType), Text)
```

Comment:
The problem is, that there's no comparison operator for the JSON datatype. So the `JSONType` attributes have to be converted to `Text` for filtering. That's basically the same as `json.dumps()`, but I'm not sure, if the ordering of the JSON elements is always identical. Maybe a JSON structure like

```
{
  "element1": "value1",
  "element2": "value2"
}
```

could be converted to `{"element1": "value1", "element2": "value2"}` by the one method, while the other method would return `{"element2": "value2", "element1": "value1"}`. This would cause the text comparison to fail, though the contents are the same. That's why I've decided to use the database casts on both sides.

But maybe I'm wrong and `json.dumps()` works exactly like casting a `JSONType` as `Text`.

Reply:
Problem is, that on my instance that creates issues with the encoding (python 2.7) ...
I'm open to any other solution getting me the symbols back and into the PDF... :)

Reply:
And why not using jsonb postgres type?

Reply:
I'm absolutely not familiar with jsonb. I just found https://stackoverflow.com/questions/22654170/explanation-of-jsonb-introduced-by-postgresql
@kdeininger @vvmruder : any opinion on this?

Reply:
I think we should continue using the `json` type.

The comparison of the legend texts is just a workaround as there is not unique type code for each legend text in the federal data until now. I as far as I know, this issue should be solved with the switch to the new specification on December 1st. After that switch, the federal data is delivered corresponding to the new model and with unique type codes. Then the legend entries can be filtered using only the type code.

As mentioned above in the answer on stackoverflow, the type `json` should be preferred, if the JSON data is only stored and queried as it is. This should be the case after the federal data has been fixed.